feat: add the number of merge buffers used for druid emitter (#16342)
apache/druid: docs/operations/metrics.md

|`query/success/count`|Number of queries successfully processed.|This metric is only available if the `QueryCountStatsMonitor` module is included.| |
|`query/failed/count`|Number of failed queries.|This metric is only available if the `QueryCountStatsMonitor` module is included.| |
|`query/interrupted/count`|Number of queries interrupted due to cancellation.|This metric is only available if the `QueryCountStatsMonitor` module is included.| |

> |`query/merge/buffersUsed`|number of merge buffers allocated to broker while performing groupBy merge queries.|This metric is only available if the `QueryCountStatsMonitor` module is included.| |

Reviewer: This shouldn't be "allocated to broker". As per my perception, allocated to the broker means the total number of merge buffers present there. Also, there is no such thing as "groupBy merge query". 
```suggestion
|`query/merge/buffersUsed`|Number of merge buffers used up to merge the results of group by queries.|This metric is only available if the `QueryCountStatsMonitor` module is included.| |
```
